agent/consul/prepared_query_endpoint.go

```diff
@@ -565,8 +570,13 @@ func (p *PreparedQuery) execute(query *structs.PreparedQuery,
 	reply.Nodes = nodes
 	reply.DNS = query.DNS
 
-	// Stamp the result for this datacenter.
-	reply.Datacenter = p.srv.config.Datacenter
+	// Stamp the result with its this datacenter or peer.
+	if peerName := query.Service.PeerName; peerName != "" {
+		reply.PeerName = peerName
+		reply.Datacenter = peerName
```

Comment:
I lean toward making Datacenter empty here when the results are from a peer, would be misleading otherwise since a peer doesn't necessarily represent a datacenter